### ISSUE TITLE
binder - baltrad

### ISSUE THREAD

**Issue by kmuehlbauer** [openradar/erad2022#10]
Several issues here:

- ~baltrad commandline tools do not work in the environment~
- baltrad2wradlib uses quite outdated/deprecated wradlib functionality
- install problems for several baltrad components
  * [x] hlhdf
  * [x] rave
  * [x] beamb
  * [x] bropo
  * [x] wrwp
  * [x] drqc
  * [x] rave_gmap   

@DanielMichelson should I make the changes directly at the source repo? https://github.com/DanielMichelson/baltrad2wradlib.git

**Comment by kmuehlbauer** [comment]
beamb is failing to build with:

```
configure: creating ./config.status
config.status: creating def.mk
mv: relocation error: /lib/x86_64-linux-gnu/libacl.so.1: symbol getxattr version ATTR_1.0 not defined in file libattr.so.1 with link time reference
config.status: error: could not create def.mk
```

**Comment by kmuehlbauer** [comment]
FYI:

It looks like `beamb` is the only failing part of baltrad. But we can't be sure, since the binder build logs are not persistent (at least I do not know how to get them). I'm giving up for now.

**Comment by kmuehlbauer** [comment]
Maybe we can export the build logs in the respective calling shell scripts?

**Comment by mgrover1** [comment]
Yeah - we need to find a better way to cache those...

**Comment by kmuehlbauer** [comment]
This is running on pangeo binder. AFAIK that is some Debian based docker container. @scollis is that correct? They use repo2docker to build the image.

I'm not sure if the error is connected with conda? Or the Python version? (3.9). We might need to think if we want to use conda compilers.

**Comment by DanielMichelson** [comment]
So I can see we're on Ubuntu 18.04 and Python 3.9, but I don't know much else yet. The build for the course in Oz 2019 used conda compilers, I think. Also, I need to ask Anders about BALTRAD and Python 3.9. I doubt there are issues, but need it confirmed.

**Comment by mgrover1** [comment]
Does baltrad require CYLP be installed?

**Comment by kmuehlbauer** [comment]
I think cylp was for pyart. But let's check this anyway

**Comment by DanielMichelson** [comment]
I don't recognize a CYLP dependency for BALTRAD. Can't find one in the BALTRAD build in https://github.com/openradar/AusOpenRadar2019/

**Comment by mgrover1** [comment]
@DanielMichelson @kmuehlbauer - is there a good way to test if  `beamb` is installed?

**Comment by DanielMichelson** [comment]
It's important that the first two BALTRAD modules are installed in the order listed in the postBuild script. I can see that HL-HDF is installed. Aside from running the unit tests that come together with each package, you can do simple imports to verify that the packages have installed. I can see that RAVE is not successfully installed yet, so none of the following packages will be either until RAVE is sorted out. 
With this in mind, I think trying to find and save the build log will be valuable to troubleshoot.

>>> import _pyhl
>>> import _rave
>>> import _beamblockage
>>> import _ropogenerator
>>> import _wrwp
>>> import GmapCreator
>>> import _dr_qc

**Comment by mgrover1** [comment]
@DanielMichelson thanks for the additional information! I am about to sign off for the weekend... I can take another look at this early next week to see if we can troubleshoot/get things running.

**Comment by kmuehlbauer** [comment]
I've tinkered a bit with it over the weekend on my fork https://github.com/kmuehlbauer/erad2022/tree/main. It looks like there is some progress.

The good news is, that it is now possible to compile baltrad on the already running binder without problems. Currently trying to get behind the issues which prevent baltrad building on binder image creation. I'll clean-up and merge this if this is resolved.

**Comment by kmuehlbauer** [comment]
With #19 baltrad is correctly installed into the environment (to be checked after build).

**Comment by kmuehlbauer** [comment]
The gmap extension does not correctly install, but might be resolved with #15.

**Comment by kmuehlbauer** [comment]
FYI: The baltrad install logs are available from the running binder in the ~/tmp/ folder.

**Comment by kmuehlbauer** [comment]
DRQC: should be fixed by #20

**Comment by kmuehlbauer** [comment]
@DanielMichelson FYI: I've made the baltrad2wradlib notebook work with the current binder state. I'll fix this at your fork.

**Comment by DanielMichelson** [comment]
Amazing work @kmuehlbauer ! I'm always happy and grateful to see someone else able to make sense out of BALTRAD.
Funny that you are devoting your energy to this now, when I was off celebrating Canada Day on Friday and @mgrover1 is off today for the 4th of July. 
I will try to take advantage of the time difference and look more closely later today.

**Comment by scollis** [comment]
+1 😊 Looking forward to trying all this out  From: Daniel Michelson ***@***.***>Date: Monday, July 4, 2022 at 9:37 AMTo: openradar/erad2022 ***@***.***>Cc: Scott ***@***.***>, Mention ***@***.***>Subject: Re: [openradar/erad2022] binder - baltrad (Issue #10)Amazing work @kmuehlbauer ! I'm always happy and grateful to see someone else able to make sense out of BALTRAD.Funny that you are devoting your energy to this now, when I was off celebrating Canada Day on Friday and @mgrover1 is off today for the 4th of July.I will try to take advantage of the time difference and look more closely later today.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by DanielMichelson** [comment]
Update: made some progress setting myself up. Forked the project and wrote a trivial Issue to confirm that I am functional again.
rave_gmap is not installing correctly due to path confusion. I will try to find a fix next.
Build logs are very spewy, but it's great that they're there! Like, why does the relevant part of the rave_gmap build start on line 3109 of the log?

**Comment by kmuehlbauer** [comment]
The logs contain every output which the install_baltrad_rave_gmap.sh script emits. The first couple thousand lines is the output of the conda activate notebook.

**Comment by kmuehlbauer** [comment]
@DanielMichelson How can we check if rave_gmap is correctly working? I've attached the according lines of the build log below. Can we see the mentioned path confusion there? AFAICT the error in the logs is due to the fact that the setup.py was tampered with to copy a certain `Lib.pth` to the relevant python folder which isn't available for some reason.

But that should be corrected by this line of code after the installation, at least that is my assesment of the situation:

https://github.com/openradar/erad2022/blob/3d76d8433c3533aaf226498c85cbb8d4d342bb9c/binder/install_baltrad_rave_gmap.sh#L32

Please note that the apache2 isn't installed and activated as in the older vagrant based courses. Maybe that's another issue.

```
+ export LD_LIBRARY_PATH=/srv/conda/envs/notebook/lib:/srv/conda/envs/notebook/hlhdf/lib:/srv/conda/envs/notebook/rave/lib
+ LD_LIBRARY_PATH=/srv/conda/envs/notebook/lib:/srv/conda/envs/notebook/hlhdf/lib:/srv/conda/envs/notebook/rave/lib
+ cd /home/jovyan/tmp
+ git clone --depth=1 https://github.com/baltrad/GoogleMapsPlugin.git
Cloning into 'GoogleMapsPlugin'...
+ cd GoogleMapsPlugin/
+ python setup.py install --prefix=/srv/conda/envs/notebook
WARNING: '' not a valid package name; please use only .-separated package names in setup.py
running install
/srv/conda/envs/notebook/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/srv/conda/envs/notebook/lib/python3.9/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
running bdist_egg
running egg_info
creating src/RAVE_Google_Maps_Plugin.egg-info
writing src/RAVE_Google_Maps_Plugin.egg-info/PKG-INFO
writing dependency_links to src/RAVE_Google_Maps_Plugin.egg-info/dependency_links.txt
writing top-level names to src/RAVE_Google_Maps_Plugin.egg-info/top_level.txt
writing manifest file 'src/RAVE_Google_Maps_Plugin.egg-info/SOURCES.txt'
reading manifest file 'src/RAVE_Google_Maps_Plugin.egg-info/SOURCES.txt'
adding license file 'LICENSE'
adding license file 'COPYING'
adding license file 'COPYING.LESSER'
writing manifest file 'src/RAVE_Google_Maps_Plugin.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib
copying src/GmapLegend.py -> build/lib
copying src/GmapCreator.py -> build/lib
copying src/GmapColorMap.py -> build/lib
copying src/GmapLayerSettings.py -> build/lib
copying src/GenerateProductInfo.py -> build/lib
copying src/googlemap_pgf_plugin.py -> build/lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
copying build/lib/GmapLegend.py -> build/bdist.linux-x86_64/egg
copying build/lib/GmapCreator.py -> build/bdist.linux-x86_64/egg
copying build/lib/GmapColorMap.py -> build/bdist.linux-x86_64/egg
copying build/lib/GmapLayerSettings.py -> build/bdist.linux-x86_64/egg
copying build/lib/GenerateProductInfo.py -> build/bdist.linux-x86_64/egg
copying build/lib/googlemap_pgf_plugin.py -> build/bdist.linux-x86_64/egg
byte-compiling build/bdist.linux-x86_64/egg/GmapLegend.py to GmapLegend.cpython-39.pyc
byte-compiling build/bdist.linux-x86_64/egg/GmapCreator.py to GmapCreator.cpython-39.pyc
byte-compiling build/bdist.linux-x86_64/egg/GmapColorMap.py to GmapColorMap.cpython-39.pyc
byte-compiling build/bdist.linux-x86_64/egg/GmapLayerSettings.py to GmapLayerSettings.cpython-39.pyc
byte-compiling build/bdist.linux-x86_64/egg/GenerateProductInfo.py to GenerateProductInfo.cpython-39.pyc
byte-compiling build/bdist.linux-x86_64/egg/googlemap_pgf_plugin.py to googlemap_pgf_plugin.cpython-39.pyc
installing package data to build/bdist.linux-x86_64/egg
running install_data
creating /srv/conda/envs/notebook/rave_gmap
creating /srv/conda/envs/notebook/rave_gmap/web
creating /srv/conda/envs/notebook/rave_gmap/web/css
copying web/css/gmlocalsearch.css -> /srv/conda/envs/notebook/rave_gmap/web/css
copying web/css/gsearch.css -> /srv/conda/envs/notebook/rave_gmap/web/css
copying web/css/lmaps.css -> /srv/conda/envs/notebook/rave_gmap/web/css
copying web/css/radar.css -> /srv/conda/envs/notebook/rave_gmap/web/css
creating /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/scl.png -> /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/logo-small-bourky.png -> /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/BALTRAD-logo-small128.png -> /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/BALTRAD-logo-small64.png -> /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/BALTRAD-logo-small.png -> /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/legendhmc.png -> /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/logo-small-chmu.png -> /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/calendar_icon.jpg -> /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/Radar2.gif -> /srv/conda/envs/notebook/rave_gmap/web/img
copying web/img/favicon.ico -> /srv/conda/envs/notebook/rave_gmap/web/img
creating /srv/conda/envs/notebook/rave_gmap/web/js
copying web/js/datetimepicker_css.js -> /srv/conda/envs/notebook/rave_gmap/web/js
copying web/js/lmaps_init.js -> /srv/conda/envs/notebook/rave_gmap/web/js
copying web/js/lmaps.js -> /srv/conda/envs/notebook/rave_gmap/web/js
copying web/js/gmap_overlay.js -> /srv/conda/envs/notebook/rave_gmap/web/js
copying web/js/radar.js -> /srv/conda/envs/notebook/rave_gmap/web/js
copying web/js/gmap_dragzoom.js -> /srv/conda/envs/notebook/rave_gmap/web/js
copying web/js/init.js -> /srv/conda/envs/notebook/rave_gmap/web/js
copying web/list_data.php -> /srv/conda/envs/notebook/rave_gmap/web
copying web/index.php -> /srv/conda/envs/notebook/rave_gmap/web
copying web/index.html -> /srv/conda/envs/notebook/rave_gmap/web
copying web/smhi-areas.xml -> /srv/conda/envs/notebook/rave_gmap/web
creating /srv/conda/envs/notebook/rave_gmap/Lib
copying src/CALIBRI.TTF -> /srv/conda/envs/notebook/rave_gmap/Lib
copying src/gmap_layer_settings.xml -> /srv/conda/envs/notebook/rave_gmap/Lib
copying README -> /srv/conda/envs/notebook/rave_gmap
copying README.ravepgf -> /srv/conda/envs/notebook/rave_gmap
copying README2 -> /srv/conda/envs/notebook/rave_gmap
copying LICENSE -> /srv/conda/envs/notebook/rave_gmap
copying COPYING -> /srv/conda/envs/notebook/rave_gmap
copying COPYING.LESSER -> /srv/conda/envs/notebook/rave_gmap
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying src/RAVE_Google_Maps_Plugin.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying src/RAVE_Google_Maps_Plugin.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying src/RAVE_Google_Maps_Plugin.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying src/RAVE_Google_Maps_Plugin.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
__pycache__.GmapLayerSettings.cpython-39: module references __file__
__pycache__.GmapLegend.cpython-39: module references __file__
creating dist
creating 'dist/RAVE_Google_Maps_Plugin-0.1-py3.9.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing RAVE_Google_Maps_Plugin-0.1-py3.9.egg
creating /srv/conda/envs/notebook/lib/python3.9/site-packages/RAVE_Google_Maps_Plugin-0.1-py3.9.egg
Extracting RAVE_Google_Maps_Plugin-0.1-py3.9.egg to /srv/conda/envs/notebook/lib/python3.9/site-packages
Adding RAVE-Google-Maps-Plugin 0.1 to easy-install.pth file

Installed /srv/conda/envs/notebook/lib/python3.9/site-packages/RAVE_Google_Maps_Plugin-0.1-py3.9.egg
Processing dependencies for RAVE-Google-Maps-Plugin==0.1
Finished processing dependencies for RAVE-Google-Maps-Plugin==0.1
Traceback (most recent call last):
  File "/home/jovyan/tmp/GoogleMapsPlugin/setup.py", line 93, in <module>
    os.rename(source, dest)  # doesn't work in some environments
FileNotFoundError: [Errno 2] No such file or directory: '/srv/conda/envs/notebook/rave_gmap/Lib.pth' -> '/srv/conda/envs/notebook/lib/python3.9/site-packages/rave_gmap.pth'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jovyan/tmp/GoogleMapsPlugin/setup.py", line 96, in <module>
    shutil.copyfile(source, dest)
  File "/srv/conda/envs/notebook/lib/python3.9/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: '/srv/conda/envs/notebook/rave_gmap/Lib.pth'
Objects created: 0
Objects deleted: 0
Objects pending: 0
+ cp web/index.html /srv/conda/envs/notebook/rave_gmap/web/.
+ rm /srv/conda/envs/notebook/rave_gmap/web/index.php
+ echo /srv/conda/envs/notebook/rave_gmap/Lib/
+ cp /home/jovyan/binder/baltrad/rave_gmap/web/smhi-areas.xml /srv/conda/envs/notebook/rave_gmap/web/.
+ cp /home/jovyan/binder/baltrad/rave_gmap/web/products.js /srv/conda/envs/notebook/rave_gmap/web/.
+ mkdir /srv/conda/envs/notebook/rave_gmap/web/data
+ cp /home/jovyan/binder/baltrad/rave_gmap/web/data/cawkr_gmaps.tgz /srv/conda/envs/notebook/rave_gmap/web/data/.
+ cd /srv/conda/envs/notebook/rave_gmap/web/data
+ tar xzf cawkr_gmaps.tgz
+ rm cawkr_gmaps.tgz
```

**Comment by DanielMichelson** [comment]
Update: I figured it out, but have not committed a solution just yet. Problem is that distutils has been deprecated and one of the arguments that GoogleMapsPlugin uses in its setup.py file ("extra_path") was removed in Python 3.8. The setup.py file was admittedly a hack anyway, probably my fault originally, but the latest developments made it unusable. I now have a setup.py file that builds successfully, but it needs to be paired with the install_baltrad_rave_gmap.sh script. This solution will need to be vetted by Anders for BALTRAD in general, but I think it will do the job for ERAD 2022. Hope to get it committed later this week.
